internal/component/pyroscope/write/write.go

```diff
@@ -318,6 +328,50 @@ func (f *fanOutClient) Append(ctx context.Context, lbs labels.Labels, samples []
 	return err
 }
 
+type PyroscopeWriteError struct {
+	StatusCode int
+}
+
+func (e *PyroscopeWriteError) Error() string {
+	return fmt.Sprintf("pyroscope write error: status %d", e.StatusCode)
+}
+
+// AppendIngest implements the pyroscope.Appender interface.
+func (f *fanOutClient) AppendIngest(ctx context.Context, profile *pyroscope.IncomingProfile) error {
+	for _, endpoint := range f.config.Endpoints {
+		u, err := url.Parse(endpoint.URL)
+		if err != nil {
+			return fmt.Errorf("parse endpoint URL: %w", err)
+		}
+
+		u.Path = path.Join(u.Path, profile.URL.Path)
+		u.RawQuery = profile.URL.RawQuery
+
+		req, err := http.NewRequestWithContext(ctx, "POST", u.String(), profile.Body)
```

Comment:
does this work if `f.config.Endpoints` has more than 1 endpoint? (I am not sure if it is possible to configure multiple endpoints)

Reply:
Yes, you can configure [multiple endpoints](https://grafana.com/docs/alloy/latest/reference/components/pyroscope/pyroscope.write/#endpoint-block). And this isn't working.
You're right, this fails when trying to read from the same body more than once which would fail.
Now, It uses a `TeeReader`, writing the data to a buffer simultaneously. For subsequent endpoints, it creates new readers from this buffer, ensuring each endpoint receives the full profile data.